djongo/sql2mongo/query.py

```diff
@@ -281,6 +292,10 @@ def _align_results(self, doc):
                             raise MigrationError(selected.column)
                         ret.append(None)
             else:
+                ## FIX: issue occurs when there's no explicit alias and we're dealing with FROM(subquery)
+                alias = str(selected.__hash__())
+                if selected.alias:  # has explicit alias
+                    alias = selected.alias
                 ret.append(doc[selected.alias])
```

Comment:
This could simplify to 
ret.append(doc[selected.alias or str(selected.__hash__())])